### DISCUSSION TITLE
Select custom component

### DISCUSSION THREAD

**Discussion by juan-manuel-alberro** [mantinedev/mantine#4876] [Q&A] [answered]
Hi all,

In Mantine v6 we had the item custom component for `Select` to render a list with a picture, like the Bender [example](https://v6.mantine.dev/core/select/#custom-item-component). 
What's the equivalent in v7? I think I explored the entire documentation but I didn't find that anymore.

Thanks!

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/combobox/?e=SelectOptionComponent

**Reply by juan-manuel-alberro** [reply]
Thank you, I wasn't able to find that in https://mantine.dev/core/select/ or https://mantine.dev/core/combobox/

**Reply by rtivital** [reply]
https://mantine.dev/core/select/#combobox
https://mantine.dev/core/combobox/#examples

**Comment by ramiel** [comment]
A question about this. If we need the select to be searchable, do we need to re-implement the search logic if we use custom components for the options?

**Reply by rtivital** [reply]
If you need your custom select component, then you need to implement logic. There are more than 50 examples available, some include the search logic.

**Reply by ramiel** [reply]
I see. I'm not saying I don't want to do this, I'm just saying that the Select component just misses this last feature of custom option render to be complete. Then, if you just need this last bit, you're forced to implement anything else (searchable, clearable and so on...). I know there are examples and they're great, it's juts that is one last feature that was even present in version 6.

At the end I'm probably need to write a "SelectWithAllFeaturesPlusCustomOptions" and reimplement everything present in the select by using the combobox 😄

**Reply by rtivital** [reply]
>  I'm just saying that the Select component just misses this last feature

Select is missing this feature for you. In my projects I do not need it, so for me the component is complete. For some people the missing feature will be creatable option, for other people the missing feature will be fuzzy search, for other people the missing feature will be custom options groupping, for other people the missing feature will be dropdown options animation, for other people the missing feature will be an option to open the dropdown inside modal instead of popover, for other people the missing feature will be to have a footer/header, for other people the missing feature will be to highlight search query based on search query...

**Reply by ramiel** [reply]
I see your point. I do not agree only because it's a regression from version 6. Anyway, good job with the new version 7, just in case this is not clear from this discussion 😄

**Reply by Aldo111** [reply]
+1 that it's a bit odd that Select doesn't have a way to do custom rendering given that the alternative is to have to reimplement the whole thing just for one feature that other libraries tend to have. Nevertheless, still a good library overall.

**Comment by Philipinho** [comment]
Mantine [v7.6.0](https://mantine.dev/changelog/7-6-0/#renderoption-prop) introduced the `renderoption` prop which solves this.
 https://mantine.dev/core/select/#renderoption
